app/controllers/articles_controller.rb

```diff
@@ -268,6 +267,15 @@ def article_params_json
     params.require(:article).permit(allowed_params)
   end
 
+  def handle_notifications(updated)
+    if updated && @article.published && @article.saved_changes["published"] == [false, true]
```

Comment:
Wouldn't the following be enough?
```suggestion
    if updated && @article.saved_changes["published"] == [false, true]
```

Reply:
Yeah probably. I decided to keep the previous code and think it over. It's probably fine to change it though.